backend/amd64/emit.mlp

```diff
@@ -732,15 +732,29 @@ let reset_probes () =
   probes := [];
   probe_semaphores := String.Map.empty
 
-let find_or_add_semaphore name =
+let find_or_add_semaphore name enabled_at_init dbg =
   match String.Map.find_opt name !probe_semaphores with
-  | Some label -> label
+  | Some (label, e) ->
+    (match e, enabled_at_init with
+     | None, None -> ()
+     | None, Some _ ->
+       let d = (label, enabled_at_init) in
+       probe_semaphores :=
+         String.Map.remove name !probe_semaphores |>
+         String.Map.add name d
+     | Some _, None ->
+       assert false
```

Comment:
(Maybe add a comment explaining
why it is unreachable.)

Reply:
added